### DISCUSSION TITLE
Ability to hide unpaired/discovered devices

### DISCUSSION THREAD

**Discussion by EnigmaCurry** [schlagmichdoch/PairDrop#104] [Ideas]
The automatic discovery feature is great, especially when getting started. But for long term pairing purposes, I do not want to see any unpaired devices from showing up in the app. Anyone on the LAN can mimic the name of another device, and unless unpaired connections are hidden, I might click on an imposter device instead of the one I meant to (apparently they can both have the same name at the same time).

So this feature might be a simple toggle to hide unpaired devices or to disable discovery altogether. At least there is the green dot to indicate the pairing so I know to click that one, this is a good start.

Am I correct in assuming the 16 digit id at the bottom of the client id never changes (unless its destroyed by the browser storage being cleared)?

**Comment by schlagmichdoch** [comment]
The 16 digit security code is a hash comprised of the webrtc encryption fingerprints of both connecting devices. So it will not necessarily stay the same on new connections but if you see the same security number on both devices you can be sure it’s the same connection and there is no MITM.

To always see all devices on the same LAN is the core feature of Snapdrop which I did not want to change so far. A possibility to hide something might make it more complex and difficult for some users.
So for now, paired devices are shown with a green dot and always in front of unpaired devices.

The plan is to also implement a public room feature to enhance the internet transfer feature (#55 ). Then, you would use the pair feature for your own devices that always discover each other and a public room (tbd whether via 7-digit or 5-letter code) to quickly send files to others over the internet. Connected devices via the room key will be marked with an (additional) yellow dot.

I mention this, as it will make a potential discovery option as you suggest more complicated and should be included into this conversation.

**Comment by schlagmichdoch** [comment]
With the freshly released version 1.8.0 the discovery and differentiation between paired and foreign devices should be clearer now.
